code-experiments/src/coco_observer.c

```diff
@@ -114,15 +113,7 @@ static int coco_observer_targets_trigger(coco_observer_targets_t *targets, const
 
     current_exponent = (int) (ceil(log10(verified_value) * number_of_targets_double));
 
-    /* If this is the first time the update was called, set the last_exponent to some value greater than the
-     * current exponent */
-    if (last_exponent == INT_MAX) {
-      last_exponent = current_exponent + 1;
-    } else {
-      last_exponent = targets->exponent;
-    }
-
-    if (current_exponent < last_exponent) {
+    if (current_exponent < targets->exponent) {
```

Comment:
It would be nice to have a comment analogous to the old one, which clarifies the initial condition of `targets->exponent`.